### PR TITLE
Docs: Update with annotations filtering and time regions

### LINKED ISSUE #67157: Annotations: Update docs for panel filtering

### PR DESCRIPTION
**Pull request by adela-almasan**
Docs update covering annotations panel filtering and the time region support.

Images used:
**Filter by panel**

screenshot-grafana-10-0-annotation-filtering

**Business hours example**

screenshot-grafana-10-0-time-regions-annotations

**Time series  - business hours example**

screenshot-grafana-10-0-timeseries-time-regions

Fixes #67157

**Special notes for your reviewer:**
@imatwawana - The Annotations docs might need a revise, I'm not sure [this](https://grafana.com/docs/grafana/latest/dashboards/build-dashboards/annotate-visualizations/) was the right place to place this bit, but also wasn't sure where else...

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: docs/sources/dashboards/build-dashboards/annotate-visualizations/index.md [hunk 1]
```diff
@@ -78,6 +78,22 @@ In Grafana v5.3+ it's possible to use template variables in the tag query. So if
 
 {{< figure src="/static/img/docs/annotations/annotation_tag_filter_variable-8-1-0.png" max-width="600px" >}}
 
+### Filter by panel
+
+You can configure to which panels to apply the annotations by choosing an option from the _Show in_ section. You can apply the annotations to `All panels` that support annotations, or filter specific panels by choosing `Selected panels` or `All panels except`. The annotations will be displayed accordingly.
```
Comment (nmarrs) [inline docs/sources/dashboards/build-dashboards/annotate-visualizations/index.md:83]:
```suggestion
You can configure which panels to apply annotations to by choosing an option from the _Show in_ section. You can apply the annotations to `All panels` that support annotations, or filter specific panels by choosing `Selected panels` or `All panels except`. The annotations will be displayed accordingly.
```

File: docs/sources/dashboards/build-dashboards/annotate-visualizations/index.md [hunk 2]
```diff
@@ -78,6 +78,22 @@ In Grafana v5.3+ it's possible to use template variables in the tag query. So if
...
+
+{{< figure src="/media/docs/grafana/screenshot-grafana-10-0-annotation-filtering.png" max-width="600px" caption="Annotation filtering" >}}
+
+### Add time region
+
+When adding or editing an annotation, you can define a repeating time region by setting _Query type_ to `Time regions` - simply define the _From_/_To_ sections with the preferred days of the week and time. You also have the possibility to change the timezone - the default being set to dashboard's timezone.
```
Comment (nmarrs) [inline docs/sources/dashboards/build-dashboards/annotate-visualizations/index.md:89]: Should we maybe mention that the datasource is -- Grafana -- here?

Comment (adela-almasan) [reply docs/sources/dashboards/build-dashboards/annotate-visualizations/index.md:89]: I think annotations are only available with grafana and gdev-testdata, so I would expect not to be that hard to find. But I might be wrong..
Maybe `Note: Time region query type is available for -- Grafana -- datasource`?

Comment (nmarrs) [reply docs/sources/dashboards/build-dashboards/annotate-visualizations/index.md:89]: good point, also the example image already shows this info too so probably ok to leave out 👍

**Review by nmarrs — APPROVED** [review]
content LGTM

**Pull request merged**